dask/dataframe/hashing.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import pandas as pd
+
+from pandas.core.categorical import is_categorical_dtype
+
+
+def hash_pandas_object(obj):
+    if isinstance(obj, (pd.Series, pd.Index)):
+        h = hash_array(obj.values).astype('uint64')
+    elif isinstance(obj, pd.DataFrame):
+        cols = obj.iteritems()
+        first_series = next(cols)[1]
+        h = hash_array(first_series.values).astype('uint64')
+        for _, col in cols:
+            h = np.multiply(h, np.uint(3), h)
+            h = np.add(h, hash_array(col.values), h)
+    else:
+        raise TypeError("Unexpected type %s" % type(obj))
+    return h
+
+
+def hash_array(vals):
+    """Given a 1d array, return an array of deterministic integers."""
+    # work with cagegoricals as ints. (This check is above the complex
+    # check so that we don't ask numpy if categorical is a subdtype of
+    # complex, as it will choke.
+    if is_categorical_dtype(vals.dtype):
+        vals = vals.codes
+
+    # we'll be working with everything as 64-bit values, so handle this
+    # 128-bit value early
+    if np.issubdtype(vals.dtype, np.complex128):
+        return hash_array(vals.real) + 23 * hash_array(vals.imag)
+
+    # MAIN LOGIC:
+
+    # First, turn whatever array this is into unsigned 64-bit ints, if we can
+    # manage it.
+    if vals.dtype == np.bool:
+        vals = vals.astype('u8')
```

Comment:
`astype` should convert, not re-view the underlying data. I just wanted to change [True True False] to [1 1 0] so that I could do math on it.